### PR TITLE
 sql: unreserve MINVALUE and MAXVALUE

### PR DESCRIPTION
**Pull request by knz**
First two commits from  #31725 and  #31731.

The keywords MINVALUE and MAXVALUE had been marked as reserved away
from being a valid column name to accommodate the CockroachDB-specific
"PARTITIONING" syntax.

This is unfortunate because these two words are perfectly fine English
names for things people may want to have as attributes in their
relations.

This patch removes the limitation by unreserving the two keywords and
handling the special words in the planning code for the partitioning
feature. In every other context the two words are not treated
specially and handled like any possible column name / identifier.

Release note (sql change): It is now again possible to use the
keywords MINVALUE and MAXVALUE as column names, for compatibility with
PostgreSQL.

### CODE REVIEW

**Review by bobvawter — APPROVED** [review]

**Comment by knz** [comment]
will want also a green light by Nikhil or Dan before merging this one.

**Review by benesch — APPROVED** [review]

**Comment by knz** [comment]
cool thanks

bors r+

**Pull request merged**
